libs/metrics/src/launch_id.rs

```diff
@@ -0,0 +1,40 @@
+//! A type to uniquely identify a process.
+//! Useful to identify process restarts in logs and metrics.
+
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use prometheus::register_int_gauge_vec;
+
+pub struct LaunchId(String);
+
+impl LaunchId {
+    pub fn generate(prefix: &str) -> Self {
+        let startup_time_millis_since_epoch: u64 = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_millis()
+            .try_into()
+            .expect("many years to come until this doesn't fit into u63");
+        let rand: u64 = rand::random();
+        let alphabet = base32::Alphabet::RFC4648 { padding: false };
+        Self(format!(
+            "{}-{}-{}",
+            prefix,
+            base32::encode(alphabet, &startup_time_millis_since_epoch.to_le_bytes()[..]),
+            base32::encode(alphabet, &rand.to_le_bytes()[..]),
+        ))
```

Comment:
Is this not some format for which we have an existing crate?

Reply:
https://github.com/mmacedoeu/rulid.rs

But do we really need all this vs. a timestamp, which could be a counter value (instead of a label)? Prometheus probably cannot handle 128-bit values.

Reply:
Hehe just checked it's 5y old.

Reply:
https://docs.rs/ulid/latest/ulid/ is more up to date :)

Reply:
ulid-rs looks all right.